src/packagedcode/dockerfileLABEL.py

```diff
@@ -0,0 +1,72 @@
+#
+# Copyright (c) nexB Inc. and others. All rights reserved.
+# ScanCode is a trademark of nexB Inc.
+# SPDX-License-Identifier: Apache-2.0
+# See http://www.apache.org/licenses/LICENSE-2.0 for the license text.
+# See https://github.com/nexB/scancode-toolkit for support or download.
+# See https://aboutcode.org for more information about nexB OSS projects.
+#
+
+
+
+import io
+from pathlib import Path
+from dockerfile_parse import DockerfileParser  
+from packagedcode import models
+from packagedcode import utils
+
+
+class DockerfileHandler(models.DatafileHandler):
+    datasource_id = 'dockerfile'
+    default_package_type = 'docker-image'
+    path_patterns = ('Dockerfile', 'containerfile', '*.dockerfile')  
+    description = 'Dockerfile (OCI) metadata handler'
+    documentation_url = 'https://docs.docker.com/engine/reference/builder/'
+
+    @classmethod
+    def parse(cls, location, package_only=False):
+        """
+        Parse a Dockerfile and yield one or more PackageData objects with OCI labels and metadata.
+        """
+        labels = cls.extract_oci_labels_from_dockerfile(location)
+
+       
+        package_data = {
+            'datasource_id': cls.datasource_id,
+            'type': cls.default_package_type,
+            'name': labels.get('name', 'unknown'),  
+            'version': labels.get('version', 'unknown'),  
+            'license_expression': labels.get('license', 'unknown'),
+            'labels': labels,  
+        }
+
+        yield models.PackageData.from_data(package_data, package_only)
+
+    @classmethod
+    def extract_oci_labels_from_dockerfile(cls, dockerfile_path):
+        """
+        Extract OCI labels from the Dockerfile using container-inspector.
+        """
+        labels = {}
+        parser = DockerfileParser()
+        parser.parse(dockerfile_path)
+        labels = parser.labels
+        return labels
+
+    @classmethod
+    def assemble(cls, package_data, resource, codebase, package_adder):

```

Comment:
I don't think we want to assemble files for these dockerfile  handlers since this is not a proper package instance that you can find in a package ecosystem, maybe we want to use `models.NonAssemblableDatafileHandler` like in https://github.com/aboutcode-org/scancode-toolkit/blob/develop/src/packagedcode/distro.py#L20 and remove this function altogether.

You can see in the issue title too: `"non-assembled" package data`